cpp/ql/src/experimental/Security/CWE/CWE-193/InvalidPointerDeref.ql

```diff
@@ -495,65 +480,89 @@ class PathSinkNode extends MergedPathNode, TPathNodeSink {
   }
 }
 
+/**
+ * A configuration that represents the full dataflow path all the way from
+ * the allocation to the dereference. We need this final dataflow traversal
+ * to ensure that the transition from the sink in `AllocToInvalidPointerConfig`
+ * to the source in `InvalidPointerToDerefFlow` did not make us construct an
+ * infeasible path (which can happen since the transition from one configuration
+ * to the next does not preserve information about call contexts).
+ */
+module FinalConfig implements DataFlow::StateConfigSig {
+  newtype FlowState =
+    additional TInitial() or
+    additional TPointerArith(PointerArithmeticInstruction pai, int delta) {
+      invalidPointerToDerefSource(_, pai, _, delta)
+    }
+
+  predicate isSource(DataFlow::Node source, FlowState state) {
+    state = TInitial() and
+    exists(DataFlow::Node derefSource |
+      invalidPointerToDerefSource(source, _, derefSource, _) and
+      InvalidPointerToDerefFlow::flow(derefSource, _)
```

Comment:
This doesn't look correct to me, as the paths we display now no longer start from an allocation.

Reply:
I think this comment is a side-effect of the misunderstanding I've tried to clear up [here](https://github.com/github/codeql/pull/13699#discussion_r1261003606): `source` is indeed the node that represents the allocation, and when we write:
```ql
invalidPointerToDerefSource(source, _, derefSource, _) and
InvalidPointerToDerefFlow::flow(derefSource, _)
```
we're saying: "the source is the node that represents the allocation, and the `derefSource` must be a node that flows to a dereference (as identified by `InvalidPointerToDerefFlow`).

Does that make sense?

Reply:
Makes sense now, and you're indeed correct with regard to the source of the confusion. Marking this as resolved.